### PR TITLE
doctrine dependencies upgrade

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| We need the dependencies to support both PHP 7.4 and 8.1
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

The following dependencies were upgraded:
- "doctrine/common": "^2.8.1" -> "^3.2.2"
- "doctrine/dbal": "^2.11" -> "^3.3.3"
- "doctrine/doctrine-bundle": "^1.8.1" -> "^2.5.7"
- "doctrine/persistence": "^1.3.7" -> "^2.4"
- "gedmo/doctrine-extensions": "^2.4.34" -> "^3.5"
- "prezent/doctrine-translatable": "^1.2" -> "^3.0"
- "prezent/doctrine-translatable-bundle": "^1.0.3" -> "^1.4"
- "shopsys/doctrine-orm": "^2.7.2" -> "doctrine/orm": "^2.11.2"
- "trikoder/oauth2-bundle": "^1.1" -> "^3.2"

### CODE REVIEW

File: packages/framework/src/Component/Paginator/QueryPaginator.php [hunk 1]
```diff
@@ -99,10 +99,9 @@ protected function getTotalNativeQuery(QueryBuilder $queryBuilder)
             $parametersAssoc[$parameter->getName()] = $parameter->getValue();
         }
 
-        list(, $flatenedParameters) = SQLParserUtils::expandListParameters(
+        $flattenedParameters = SqlParametersFlattener::flattenArrayParameters(
```
Comment (grossmannmartin) [inline packages/framework/src/Component/Paginator/QueryPaginator.php:102]: 🤔 is this native query necessary at all? 
this whole implementation is quite uncommon. Can we discuss it please?

File: upgrade/UPGRADE-v9.2.0-dev.md [hunk 2]
```diff
@@ -8,6 +8,22 @@ There you can find links to upgrade notes for other versions too.
 ## Composer dependencies
 - remove `vasek-purchart/console-errors-bundle` dependency ([#2408](https://github.com/shopsys/shopsys/pull/2408))
     - see #project-base-diff to update your project
+- upgrade your dependencies to the versions that support both PHP 7.4 and 8.1 ([#2417](https://github.com/shopsys/shopsys/pull/2417))
+    - see #project-base-diff to update your project
+    - make sure you check all the upgrading notes of the upgraded packages and adjust your codebase appropriately
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v9.2.0-dev.md:13]: maybe links to the upgrade notes could be helpful?

File: upgrade/UPGRADE-v9.2.0-dev.md [hunk 3]
```diff
@@ -8,6 +8,22 @@ There you can find links to upgrade notes for other versions too.
...
+        + public function __construct(DatabaseConnectionCredentialsProvider $databaseConnectionCredentialsProvider)
+        ```
+    - `EntityExtensionSubscriber::getClassMetadataForEntity` changed its interface:
+        ```diff
+        - protected function getClassMetadataForEntity(string $entityClass): ClassMetadataInfo
+        + protected function getClassMetadataForEntity(string $entityClass): ClassMetadata
+        ```
+    - `Admin/AdvertController` has a new dependency on `EntityManagerInterface` in the constructor
+    - `Admin/BrandController` has a new dependency on `EntityManagerInterface` in the constructor
+    - `Admin/SliderController` has a new dependency on `EntityManagerInterface` in the constructor
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v9.2.0-dev.md:26]: Is removal of the parameter from the flush method worth mentioning?

File: upgrade/UPGRADE-v9.2.0-dev.md [hunk 4]
```diff
@@ -8,6 +8,22 @@ There you can find links to upgrade notes for other versions too.
 ## Composer dependencies
 - remove `vasek-purchart/console-errors-bundle` dependency ([#2408](https://github.com/shopsys/shopsys/pull/2408))
     - see #project-base-diff to update your project
+- upgrade your dependencies to the versions that support both PHP 7.4 and 8.1 ([#2417](https://github.com/shopsys/shopsys/pull/2417))
```
Comment (grossmannmartin) [inline upgrade/UPGRADE-v9.2.0-dev.md:11]: does it make sense to write it less generic (in PR description and title too). 
I think it's about doctrine related packages, what do you think?

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: project-base/config/packages/nyholm_psr7.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    Http\Message\StreamFactory:
+        alias: nyholm.psr7.httplug_factory
+    Http\Message\UriFactory:
+        alias: nyholm.psr7.httplug_factory
+
+    nyholm.psr7.psr17_factory:
+        class: Nyholm\Psr7\Factory\Psr17Factory
+
+    nyholm.psr7.httplug_factory:
+        class: Nyholm\Psr7\Factory\HttplugFactory
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply project-base/config/packages/nyholm_psr7.yaml:32]: TBH, I do not know, this config was added automatically after the installation of the package and I did not have the guts to change it (just our coding standards changed the notation from `@...` to `alias:...` :smile:

File: packages/framework/src/Component/Doctrine/DatabaseSchemaFacade.php [hunk 6]
```diff
@@ -56,7 +56,7 @@ public function importDefaultSchema()
 
         $line = fgets($handle);
         while ($line !== false) {
-            $connection->query($line);
+            $connection->executeQuery($line);
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply packages/framework/src/Component/Doctrine/DatabaseSchemaFacade.php:59]: ah, bad Rosťa...fixed in 79f5308bb

File: project-base/tests/App/Test/Codeception/Helper/DatabaseHelper.php [hunk 7]
```diff
@@ -19,34 +19,27 @@ public function _initialize()
...
         ]);
     }
 
     /**
-     * @param \Doctrine\DBAL\Connection $connection
+     * @param \Shopsys\FrameworkBundle\Component\Doctrine\DatabaseConnectionCredentialsProvider $databaseConnectionCredentialsProvider
      * @return string
      */
-    private function getConnectionDsn(Connection $connection)
+    private function getConnectionDsn(DatabaseConnectionCredentialsProvider $databaseConnectionCredentialsProvider): string
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply project-base/tests/App/Test/Codeception/Helper/DatabaseHelper.php]: sure, good idea, done in 304ab3f34

File: packages/framework/src/Component/EntityExtension/EntityExtensionSubscriber.php [hunk 8]
```diff
@@ -213,11 +214,11 @@ protected function isOneOfRegisteredEntities(string $entityClass): bool
...
     /**
      * @param class-string $entityClass
-     * @return \Doctrine\ORM\Mapping\ClassMetadataInfo
+     * @return \Doctrine\ORM\Mapping\ClassMetadata
      */
-    protected function getClassMetadataForEntity(string $entityClass): ClassMetadataInfo
+    protected function getClassMetadataForEntity(string $entityClass): ClassMetadata
     {
-        $classMetadata = new ClassMetadataInfo(
+        $classMetadata = new ClassMetadata(
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply packages/framework/src/Component/EntityExtension/EntityExtensionSubscriber.php:220]: yes - ce0e22de2 :+1:

File: packages/framework/src/Component/Paginator/QueryPaginator.php (hunk 1 above)
Comment (vitek-rostislav) [reply packages/framework/src/Component/Paginator/QueryPaginator.php:102]: ok, let's discuss it :slightly_smiling_face:

File: upgrade/UPGRADE-v9.2.0-dev.md (hunk 3 above)
Comment (vitek-rostislav) [reply upgrade/UPGRADE-v9.2.0-dev.md:26]: I do not think so - in such a case, I would feel like I need to mention all the other changes as well, I believe the combination of the project-base diff along with all the packages upgrade notes is sufficient

File: upgrade/UPGRADE-v9.2.0-dev.md (hunk 2 above)
Comment (vitek-rostislav) [reply upgrade/UPGRADE-v9.2.0-dev.md:13]: I was afraid there will be too much of them, but it is quite ok, d20ea16f7 :slightly_smiling_face:

File: upgrade/UPGRADE-v9.2.0-dev.md (hunk 4 above)
Comment (vitek-rostislav) [reply upgrade/UPGRADE-v9.2.0-dev.md]: indeed - d20ea16f7 :slightly_smiling_face:

File: project-base/composer.json [hunk 9]
```diff
@@ -54,11 +55,11 @@
         "jms/serializer-bundle": "^2.4",
         "jms/translation-bundle": "1.4.4",
         "league/flysystem": "^1.1.4",
+        "nyholm/psr7": "^1.5",
         "phing/phing": "^2.16.1",
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply project-base/composer.json:59]: ah, damn, the approach does not help, let's discuss this, please :pray:

[the remark this answers is not in the document]
Comment (grossmannmartin) [reply project-base/composer.json:59]: I just run composer update in project base folder :D

File: packages/framework/src/Migrations/Version20190521071335.php [hunk 10]
```diff
@@ -13,7 +14,7 @@ class Version20190521071335 extends AbstractMigration
     public function up(Schema $schema): void
     {
         $this->sql('ALTER TABLE products ADD uuid UUID DEFAULT NULL');
-        $this->sql('UPDATE products SET uuid = uuid_generate_v4()');
+        $this->sql('UPDATE products SET uuid = :uuid', ['uuid' => Uuid::uuid4()->toString()]);
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/framework/src/Migrations/Version20190521071335.php:17]: can you please mention this in the upgrade notes? 😇

File: project-base/composer.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply project-base/composer.json:59]: updated

**Review by grossmannmartin — APPROVED** [review]
nice work 😍

File: packages/framework/src/Component/Cron/CronModuleFacade.php [hunk 11]
```diff
@@ -141,7 +141,7 @@ public function markCronAsFailed(CronModuleConfig $cronModuleConfig): void
         $cronModule = $this->cronModuleRepository->getCronModuleByServiceId($cronModuleConfig->getServiceId());
         $cronModule->setStatusFailed();
 
-        $this->em->flush($cronModule);
+        $this->em->flush();
```
Comment (pk16011990) [inline packages/framework/src/Component/Cron/CronModuleFacade.php:144]: method `markCronAsFailed()` is called on any throwable and this flush() without parameter can save some invalid state of entities. Is it safe and acceptable?

Comment (vitek-rostislav) [reply packages/framework/src/Component/Cron/CronModuleFacade.php]: true indeed, rewritten in 63d6f7432

**Pull request merged**